src/main/java/frc/robot/util/AutoLogic.java

```diff
@@ -0,0 +1,248 @@
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.Controls;
+import frc.robot.Robot;
+import frc.robot.Subsystems;
+import frc.robot.subsystems.drivebase.CommandSwerveDrivetrain;
+import java.io.IOException;
+import java.util.List;
+import org.json.simple.parser.ParseException;
+
+public class AutoLogic {
+  private static final Robot r = Robot.getInstance();
+  private static final Subsystems s = r.subsystems;
+  private static final Controls controls = r.controls;
+
+  private static SendableChooser<String> autoPicker = new SendableChooser<String>();
+  private static SendableChooser<String> autoAction = new SendableChooser<String>();
+  private static SendableChooser<String> testedAutos = new SendableChooser<String>();
+
+  private static ShuffleboardTab tab = Shuffleboard.getTab("Autos");
+
+  public static void configureAuto(CommandSwerveDrivetrain drivebase) {
+
+    try {
+      AutoBuilder.configure(
+          () -> drivebase.getState().Pose, // Robot pose supplier
+          (pose) ->
+              drivebase.resetPose(
+                  pose), // Method to reset odometry (will be called if your auto has a starting
+          // pose)
+          () -> drivebase.getState().Speeds, // ChassisSpeeds supplier. MUST BE ROBOT RELATIVE
+          (speeds, feedforwards) ->
+              drivebase.setControl(
+                  new SwerveRequest.ApplyRobotSpeeds()
+                      .withSpeeds(speeds)), // Method that will drive the robot given ROBOT RELATIVE
+          // ChassisSpeeds. Also optionally outputs individual module
+          // feedforwards
+          new PPHolonomicDriveController( // PPHolonomicController is the built in path following
+              // controller for holonomic drive trains
+              new PIDConstants(5.0, 0.0, 0.0), // Translation PID constants
+              new PIDConstants(5.0, 0.0, 0.0) // Rotation PID constants
+              ),
+          RobotConfig.fromGUISettings(), // The robot configuration
+          () -> {
+            // Boolean supplier that controls when the path will be mirrored for the red alliance
+            // This will flip the path being followed to the red side of the field.
+            // THE ORIGIN WILL REMAIN ON THE BLUE SIDE
+
+            var alliance = DriverStation.getAlliance();
+            if (alliance.isPresent()) {
+              return alliance.get() == DriverStation.Alliance.Red;
+            }
+            return false;
+          },
+          s.drivebaseSubsystem // Reference to this subsystem to set requirements
+          );
+
+    } catch (IOException | ParseException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
```

Comment:
Would be good to also call `DriverStation.reportError` to more visibly surface the error